File: bench/native/fib.c
// bench: recursive fib(30) — C reference (self-timed like `wide --time`).
#include <stdio.h>
#include <time.h>

long long fib(long long n) { return n < 2 ? n : fib(n - 1) + fib(n - 2); }

int main(void) {
    clock_t t0 = clock();
    long long r = fib(30);
    double ms = (double)(clock() - t0) * 1000.0 / CLOCKS_PER_SEC;
    printf("%lld\n", r);
    fprintf(stderr, "time: %.3f ms\n", ms);
    return 0;
}

File: bench/native/fib.rs
// bench: recursive fib(30) — Rust reference (self-timed like `wide --time`).
fn fib(n: i64) -> i64 {
    if n < 2 { n } else { fib(n - 1) + fib(n - 2) }
}

fn main() {
    let t0 = std::time::Instant::now();
    let r = fib(30);
    let ms = t0.elapsed().as_secs_f64() * 1000.0;
    println!("{}", r);
    eprintln!("time: {:.3} ms", ms);
}
